docs(weave): add .delete() to object doc page (#3452)
wandb/weave: docs/docs/guides/tracking/objects.md

  </TabItem>
</Tabs>

## Deleting an object

<Tabs groupId="programming-language" queryString>
  <TabItem value="python" label="Python" default>

>     You can call `.delete()` on any object Ref to delete that version of the object.

Reviewer: ```suggestion
     To delete a version of an object, call `.delete()` on the object reference.
```

Author: `Ref` is sort of a special word, I think we should keep it (see section below), committing the other copy though.
